### PR TITLE
Make MDC DataCite URl configurable

### LINKED ISSUE #6137: MakeDataCountApi is hardcoded to production datacite server
MakeDataCountApi, on line 142, hardcodes the api.datacite.org server. It might be better if it leveraged the doi.baseurlstring jvm option so that development/test servers using doi.baseurlstring=https://mds.test.datacite.org would query api.test.datacite.org for citations. (I verified that DOIs minted at mds.test.datacite.org report their info from api.test.datacite.org and not api.datacite.org, the latter of which just returns an ~empty json doc.)

Since the doi.baseurlstring currently includes the protocol and the mds service, I'm not sure whether it would be better to add a second doi.MDCurlstring or to switch to making the server configurable and prepending hardcoded https://mds. or https://api.  where needed. (Or use a Dataverse setting since there is no username/password associated with the public api service?)

### PR DESCRIPTION
**Pull request by qqmyers**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Note that we use the "closes" syntax below to trigger Github's automation to close the corresponding issue once the pull request is merged.

Thanks for your contribution to Dataverse!

## Related Issues

- closes #6137 MakeDataCountApi is hardcoded to production datacite server

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: doc/sphinx-guides/source/installation/config.rst [hunk 1]
```diff
@@ -956,6 +957,28 @@ See also these related database settings below:
 - :ref:`:Authority`
 - :ref:`:Shoulder`
 
+.. _doi.baseurlstring:
```
Comment (djbrooke) [inline doc/sphinx-guides/source/installation/config.rst]: Should this be doi.mdcbaseurlstring instead?

Comment (qqmyers) [reply doc/sphinx-guides/source/installation/config.rst]: yep

**Comment by kcondon** [comment]
@qqmyers @pdurbin 

./install
String found where operator expected at ./install line 146, near "'DOI_MDCBASEURL'"
	(Missing semicolon on previous line?)
String found where operator expected at ./install line 175, near "'DOI_MDCBASEURL'"
	(Missing semicolon on previous line?)
syntax error at ./install line 146, near "'DOI_MDCBASEURL'"
syntax error at ./install line 175, near "'DOI_MDCBASEURL'"
Execution of ./install aborted due to compilation errors.

**Comment by pdurbin** [comment]
run tests

**Comment by kcondon** [comment]
I have built and checked the docs, tested the endpoints. Had trouble running the installation script -think it may just be missing a , when adding an addition element to list. Have not checked whether Dataverse actually uses the new MDC setting.

**Comment by pdurbin** [comment]
Yeah, it looks like there may be two places in the Perl script where commas are missing.

**Comment by djbrooke** [comment]
Thanks @qqmyers. Ready for us to take another look?

**Comment by qqmyers** [comment]
@djbrooke - Yep. BTW - When the fix is pretty obvious as it was here (two commas), you guys should feel free to make the change and keep going. (You're welcome to edit any time but here it definitely seems like more effort to document the issue, reassign, wait, etc.) I'm happy to support PRs when that makes less work for you guys, but no need for more work either.

**Comment by djbrooke** [comment]
Thanks @qqmyers, I'll try to do a better job keeping on top of when small changes need to be done and having us do them.

**Comment by pdurbin** [comment]
@qqmyers I'm going to work on one such small change. @kcondon requested some logging.

Also, a little bird told me that there's a bug in the changes to the Perl script. 😄 

Thanks, as always, for hacking on Dataverse! 🎉

**Comment by kcondon** [comment]
@qqmyers @pdurbin 

installer:
-JVM option name transposed: mdcbaseurl is set as basemdcurl instead.
-Default installation value for mdcbaseurl is not set (empty), need to also set the gfish env value.

Jim, are you able to test run the installer on your end?

Phil, thanks for adding the logger so I can check which url is used.

**Comment by qqmyers** [comment]
@kcondon - just updated the naming for the jvm option. (the api code was also different). I'm not really able to run the installer - really working on our branch and using the auto build/deploy process we have so I'm really guessing on changes to the various scripts. 
W.r.t. a default value - I made it backwards compatible, so if you don't set it, the hardcoded value from earlier versions are used. If that's removed, the update instructions will need to include the step to set it.

**Comment by qqmyers** [comment]
@pdurbin - thanks for adding logging, and, as for hacking on Dataverse, at least you get a T-shirt this month :-)

**Comment by landreev** [comment]
For the main installer script to work, could you please add the following 2 lines to scripts/installer/install, at line 1104: 

```
    $ENV{'DOI_BASEURL'} = $CONFIG_DEFAULTS{'DOI_BASEURL'};
    $ENV{'DOI_MDCBASEURL'} = $CONFIG_DEFAULTS{'DOI_MDCBASEURL'};
```

(it was missing it for the DOI_BASEURL variable too, yes;)

I would do it myself, but the branch is in your repository. 
Thanks.

**Comment by pdurbin** [comment]
> I would do it myself, but the branch is in your repository.

Yeah, I got "ERROR: Permission to QualitativeDataRepository/dataverse.git denied to pdurbin." when I tried to `git push QualitativeDataRepository IQSS/6137` so I just created this pull request: https://github.com/QualitativeDataRepository/dataverse/pull/26

I haven't looked at the Perl script yet and don't plan to unless someone wants me to.

**Comment by pdurbin** [comment]
Sorry, it is WAY to easy to accidentally close a pull request. Re-opened! I mis-clicked. 😞

**Comment by pdurbin** [comment]
@kcondon in order to bump up the logging to see "Requesting citations from https://api.datacite.org/events?doi=10.5072/FK2/BL2IBM&source=crossref" (or whatever) in server.log, you can run this:

asadmin set-log-levels edu.harvard.iq.dataverse.api.MakeDataCountApi=FINE

**Comment by qqmyers** [comment]
Hmm - it's supposed to work. When I create the PRs I check the box to allow editing as in 
image
This image is a test where I was tried our v4.17-qdr branch and you can see it says people with write on IQSS/dataverse (not QDR/dataverse) can commit to the branch in the PR. Could it be only by editing in the github website?

**Comment by djbrooke** [comment]
@qqmyers FWIW I tried update two weeks ago what I eventually mentioned in https://github.com/IQSS/dataverse/pull/6265#discussion_r333280134 and was also denied. I assumed it was due to me being not good at Github. :) Sorry, I should have raised the flag then.

**Pull request merged**